alda-mode.el

```diff
@@ -54,19 +56,83 @@ If you are experiencing problems, try clearing your history with 'alda-history-c
   "Alda customization options"
   :group 'applications)
 
+(defgroup alda-mode-inf
+  nil
+  "Mode to interact with a Alda interpreter."
+  :group 'Alda
+  :tag "Inferior Alda")
+
 (defcustom alda-binary-location nil
   "Alda binary location for `alda-mode'.
 When set to nil, will attempt to use the binary found on your $PATH.
 This must be a _full_ path to your alda binary."
   :type 'string
   :group 'Alda)
 
+(defcustom alda-interpreter "alda repl"
+  "The interpreter that `run-alda' should run. This should
+ be a program in your PATH or the full pathname of the alda interpreter."
+  :type 'string
+  :group 'alda-mode-inf)
+
+(defconst alda-inf-buffer-name "*inferior-alda*")
+
+(define-derived-mode alda-mode-inf comint-mode "Inferior Alda"
+  "Major mode for interacting with a Alda interpreter.
+
+\\{inferior-alda-mode-map\\}"
+  (define-key alda-mode-inf-map [(meta return)] 'comint-accumulate)
+
+  ;; Comint configuration
+  (make-local-variable 'comint-input-sender)
+  (setq comint-input-sender 'alda-input-sender))
+
+(defun alda-input-sender (proc string)
+  (comint-send-string proc string)
+  ;; (comint-send-string proc "\nemacs:end\n")) ;; Heineman's contrib (06/03/2007)
+  (comint-send-string proc "\n"))
+
+;;;###autoload
+(defun alda-interpreter-running-p-1 ()
+  ;; True iff a Alda interpreter is currently running in a buffer.
+  (comint-check-proc alda-inf-buffer-name))
+
+(defun alda-check-interpreter-running ()
```

Comment:
I really dislike this name, it implies it returns whether the interpreter is running or not (and does no action). Can we name it `alda-interpreter-start` perhaps (I'm pretty bad at names, so if you can think of something better, use that instead).

Reply:
Good point. renamed to alda-check-or-start-interpreter